src/pages/Notes/components/NoteForm.jsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import { Box, TextField, Button, Collapse, Paper } from "@mui/material";
+
+import { useDialogActions } from "../../../hooks/useDialogActions";
+
+function NoteForm() {
+  const [expanded, setExpanded] = useState(false);
+  const [title, setTitle] = useState("");
+  const [content, setContent] = useState("");
+  const { openAddDialog } = useDialogActions();
+
+  const handleCancel = () => {
+    setExpanded(false);
+    setTitle("");
+    setContent("");
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!title || !content) return;
+    openAddDialog({title, content})
+  };
+
+  return (
+    <Paper sx={{ p: 2, mt: 2 }}>
+      <Collapse in={!expanded}>
+        <Box onClick={() => setExpanded(true)} sx={{ cursor: "pointer" }}>
+          Click here to add a new note...
+        </Box>
+      </Collapse>
+      <Collapse in={expanded}>
+        <form onSubmit={handleSubmit}>
+          <TextField
+            label="Title"
+            fullWidth
+            margin="normal"
+            value={title}
+            onChange={(e) => setTitle(e.target.value)}
+          />
+          <TextField
+            label="Content"
+            fullWidth
+            multiline
+            rows={4}
+            margin="normal"
+            value={content}
+            onChange={(e) => setContent(e.target.value)}
```

Comment:
we have two ways better than this one 
1- we can combine the two states together if they are related and use the function below.
2- we can create the handlers functions like this `handleContentChange` so we can expand and add on it in the future we are not limited to one line in most of the cases.

```suggestion
const handleChange =(e)=>{
setContent(prev=>([...prev,[e.target.name]:e.target.value])
}

name="content"
 value={content}
onChange={handleChange}
```